### PR TITLE
update k8s clusters to create all nodes using the same flist

### PR DESCRIPTION
**Pull request by Eslam-Nawara**
### Description

update k8s clusters to create all nodes using the same flist

### Changes

update k8s clusters to create all nodes using the same flist

### Related Issues

- #1158 

### Checklist

- [x] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstring

### Tests:
- `Grid client integration tests`: https://github.com/threefoldtech/tfgrid-sdk-go/actions/runs/10680266593

### CODE REVIEW

File: grid-client/workloads/k8s.go [hunk 1]
```diff
@@ -197,6 +202,33 @@ func (k *K8sCluster) ValidateIPranges() error {
 	}
 
 	for _, w := range k.Workers {
+		// // ValidateFlist validate check sums for k8s flist
```
Comment (rawdaGastan) [inline grid-client/workloads/k8s.go]: shouldn't it be removed?

File: grid-client/workloads/k8s.go [hunk 2]
```diff
@@ -298,3 +305,18 @@ func (k *K8sNode) zosWorkload(cluster *K8sCluster, isWorker bool) (K8sWorkloads
 
 	return K8sWorkloads
 }
+
+func (k *K8sCluster) validateFlist() error {
```
Comment (rawdaGastan) [inline grid-client/workloads/k8s.go]: I think we validate the flists for master and workers, this function assigns the flist

Comment (Eslam-Nawara) [reply grid-client/workloads/k8s.go]: it also validate that either the cluster flist or the master flist exists

Comment (rawdaGastan) [reply grid-client/workloads/k8s.go]: I think we should validate the cluster flist only, others are validated,
Also we should display warning for example if cluster, master and workers flists don't match, what do you think?

**Review by rawdaGastan — APPROVED** [review]

**Pull request merged**